File: Ride_Sharing_App/src/main/java/main/ride_sharing_app/websocket/dto/OrderNotification.java
package main.ride_sharing_app.websocket.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderNotification {
    private Long orderId;
    private String startLocation;
    private String endLocation;
    private Double estimatedPrice;
    private Double distance;
    private Long timestamp;
    private String status;
    private String message;
} 

File: Ride_Sharing_UI/src/types/OrderNotification.ts
import { DriverInfoDTO } from "./DriverInfoDTO";

export interface OrderNotification {
  orderId: number;
  status: string;
  message: string;
  timestamp: number;
  driverInfo?: DriverInfoDTO;
  estimatedArrival?: number;
  startLocation?: string;
  endLocation?: string;
  estimatedPrice?: number;
  distance?: number;
}
